test/ci-ecr-stack.test.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import {
+  expect, countResources, haveResourceLike, ResourcePart,
+} from '@aws-cdk/assert';
+import { App, RemovalPolicy } from '@aws-cdk/core';
+import { CiEcrStack } from '../lib/ci-ecr-stack';
+import { DeployAssetsProps, DeployAwsAssets } from '../lib/deploy-aws-assets';
+
+test('ECR Stack with Resources', () => {
+  const app = new App({
+    context: { useSsl: 'true', runWithOidc: 'true' },
+  });
+
+  // WHEN
+  const deployAssetsProps : DeployAssetsProps = {
+    removalPolicy: RemovalPolicy.DESTROY,
+    mainNodeAccountNumber: '99999999',
+    envName: 'dev',
+    env: {
+      region: 'us-east-1',
+    },
+    deployECR: false,
+    repositories: [
+      'opensearch',
+      'opensearch-dashboards',
+    ],
+  };
+
+  const deployAwsAssetsStack = new DeployAwsAssets(app, 'TestStack', deployAssetsProps);
+
+  const ecrStack = new CiEcrStack(deployAwsAssetsStack, 'ecrStack', deployAssetsProps);
+
+  // THEN
+  expect(ecrStack).to(countResources('AWS::IAM::Role', 1));
+  expect(ecrStack).to(countResources('AWS::IAM::ManagedPolicy', 1));
+});
+
+test('ECR Stack No ECR Repositories', () => {
+  const app = new App({
+    context: { useSsl: 'true', runWithOidc: 'true' },
+  });
+
+  // WHEN
+  const deployAssetsProps : DeployAssetsProps = {
+    removalPolicy: RemovalPolicy.DESTROY,
+    mainNodeAccountNumber: '99999999',
+    envName: 'dev',
+    env: {
+      region: 'us-east-1',
+    },
+    deployECR: false,
+  };
+
+  const deployAwsAssetsStack = new DeployAwsAssets(app, 'TestStack', deployAssetsProps);
+
+  const ecrStack = new CiEcrStack(deployAwsAssetsStack, 'ecrStack', deployAssetsProps);
+
+  // THEN
+  expect(ecrStack).to(countResources('AWS::ECR::PublicRepository', 0));
+  expect(deployAwsAssetsStack).to(countResources('AWS::ECR::PublicRepository', 0));
+  expect(ecrStack).to(countResources('AWS::IAM::Role', 1));
+  expect(ecrStack).to(countResources('AWS::IAM::ManagedPolicy', 1));
+  expect(deployAwsAssetsStack).to(countResources('AWS::CloudFormation::Stack', 1));
```

Comment:
I don't think this test case is required.

Reply:
It is. This test case verifies the behavior of the code if the repositories are not specified.

Reply:
It is. This test case verifies the behavior of the code if the repositories are not specified.